### PR TITLE
Perfusion-related fixes (PHI, PerfusionPCA, GUI/CLI)

### LINKED ISSUE #1450: Crash on using "Apply Mask" from drawing tab
**Describe the bug**
CaPTk crashes without an error message when hitting "Apply Mask" from the drawing tab of the GUI. 

Witnessed on CaPTk 1.8.1, 1.8.2.Alpha on Windows.

**Exepcted behavior**
Either apply the mask as expected, or crash with some kind of error message if unavoidable -- even if it's just telling users to submit a bug report.

### LINKED ISSUE #1472: [1.9.0.Alpha] Cannot load 4D images and 3D images together
**Describe the bug**
When loading 4D images alongside 3D images, e.g. for PHI which requires a perfusion and a T1-GD image, the load fails with the sanity check failure message. This occurs regardless of the order of loading.

Additionally, very odd origins are shown when the 4D image is loaded alone.

### LINKED ISSUE #1473: [Feature Request] batch mode for PHI, set a specific number of time points
Initially requested by Spyros.

Add batch mode functionality to PHI  so that analysts can run it across multiple subjects. Also enable a time points parameter that allows a user to specify an amount of time points to use in processing. Warn users about the time points being used in general.

### LINKED ISSUE #1474: [Feature Request] 4D Slice indicator
Initially requested by Spyros/Hamed.

Add an indicator next to the 4D scroll bar that tells users what time-slice they are currently on and how many there are total.

### LINKED ISSUE #1475: [1.9.0.Alpha] PerfusionPCA needs more clarity on file reqs.
Initially reported by Hamed.

PerfusionPCA fails with unhelpful messages when a user selects a directory that doesn't exactly match the structure. To make this clearer, specify the requirements clearly on the CLI and GUI.

### PR DESCRIPTION
**Pull request by AlexanderGetka-cbica**
Fixed #1472 
Fixed #1473 
Fixed #1474 
Fixed #1475
Seems to have also fixed #1450 

Summary:
Adds a text indicator of 4D slice number to the slider bar.
Clarifies perfusionPCA filename requirements (but doesn't make them more intelligent).
Fixes loading of 4D and 3D images together.
Adds batch mode and time-points parameters to PHI (EGFRvIIISurrogateIndex).

Also disables self-host mac builds in the Azure pipeline. @sarthakpati may need to configure the requirements on GitHub side, not sure about that.

### CODE REVIEW

**Comment by sarthakpati** [comment]
Are we trying to fix Codacy issues, as well? For e.g.:

image

**Comment by AlexanderGetka-cbica** [comment]
> Are we trying to fix Codacy issues, as well? For e.g.:
> 
> image

Ah, yeah -- I should've caught that. My bad. I won't change the code-style related ones for this but I'll address any "medium" or above.

**Review by sarthakpati — APPROVED** [review]
~LGTM!~

Accidentally put this on.

File: src/view/gui/fMainWindow.cpp [hunk 1]
```diff
@@ -1823,10 +1823,28 @@ void fMainWindow::LoadSlicerImages(const std::string &fileName, const int &image
...
+          for (int j = 0; j < 3; j++)
+          {
+              originalDirectionIn3D[i][j] = originalDirection[i][j];
+          }
+      }
+      imageManager->SetOriginalDirection(originalDirectionIn3D);
+      */
+      //auto tempImage = cbica::GetImageOrientation< ImageTypeFloat4D >(imagePerf);
+      //imageManager->SetOriginalOrientation(tempImage.first);
+      //imagePerf = ChangeImageDirectionToIdentity< ImageTypeFloat4D >(tempImage.second);
```
Comment (sarthakpati) [inline src/view/gui/fMainWindow.cpp:1845]: Any reason for this commented code to be present? If yes, then please place a comment here.

**Review by sarthakpati — CHANGES_REQUESTED** [review]

**Comment by AlexanderGetka-cbica** [comment]
I'll remove that commented code in the PR that updates the docs and version. Thanks @sarthakpati !

**Pull request merged**